compendium/test_bundles/TestBundleDSCA10/src/ServiceImpl.cpp

```diff
@@ -0,0 +1,37 @@
+#include "ServiceImpl.hpp"
+#include "cppmicroservices/Bundle.h"
+#include "cppmicroservices/cm/ConfigurationAdmin.hpp"
+#include <iostream>
+
+namespace sample
+{
+
+    ServiceComponentCA10::ServiceComponentCA10(std::shared_ptr<cppmicroservices::AnyMap> const& props)
+        : properties(props)
+    {
+    }
+    void
+    ServiceComponentCA10::Modified(std::shared_ptr<ComponentContext> const& /*context*/,
+                                   std::shared_ptr<cppmicroservices::AnyMap> const& configuration)
+    {
+        std::lock_guard<std::mutex> lock(propertiesLock);
+        properties = configuration;
+        auto CA = cppmicroservices::any_cast<std::shared_ptr<cppmicroservices::service::cm::ConfigurationAdmin>>(
+            properties->at("CA"));
+
+        auto configObject = CA->GetConfiguration(cppmicroservices::any_cast<std::string>(properties->at("configID")));
```

Comment:
Seems a funny way to get hold of `ConfigurationAdmin`, why not use a reference & constructor injection?

Reply:
Indeed.